app/src/org/commcare/models/database/DbUtil.java

```diff
@@ -94,7 +93,9 @@ public static PrototypeFactory getPrototypeFactory(Context c) {
                         if (!emptyc) {
                             continue;
                         }
-                        classNames.add(cn);
+                        if (Externalizable.class.isAssignableFrom(prototype)) {
+                            classNames.add(cn);
+                        }
```

Comment:
Realizing, now that I understand the prototype factory code better, that https://github.com/dimagi/commcare-odk/pull/1050/files#r51792765 deleted the wrong `classNames.add(cn)` code